python/cudf/cudf/core/dataframe.py

```diff
@@ -1218,13 +1218,41 @@ def _slice(self: T, arg: slice) -> T:
             return self
         start, stop, stride = arg.indices(num_rows)
 
+        # early stop for empty cases
+        if len(range(start, stop, stride)) == 0:
+            columns = ColumnAccessor(
+                {
+                    colname: column.column_empty_like(col, newsize=0)
+                    for colname, col in self._data.items()
+                },
+                multiindex=self._data.multiindex,
+                level_names=self._data.level_names,
+            )
+
+            if isinstance(self.index, MultiIndex):
+                mi_columns = ColumnAccessor(
+                    {
+                        colname: column.column_empty_like(col, newsize=0)
+                        for colname, col in self.index._data.items()
+                    }
+                )
+                index = MultiIndex._from_data(mi_columns)
+            else:
+                index = (
+                    RangeIndex(start=0, stop=0)
+                    if start < stop
+                    else Index([], dtype=self.index.dtype)
```

Comment:
I assume this is to mimic pandas behavior?

Reply:
This is to match pandas yes. I actually had the rule slightly wrong here. The rule appears to be that when returning an empty slice from a `RangeIndex`'ed dataframe, a length zero range index is produced that spans from `start` to `stop`. 

```python
>>> df = pd.DataFrame({'a':[1,2,3], 'b':[4,5,6]})
>>> df[3:2:2].index
RangeIndex(start=3, stop=2, step=2)
```

However other kinds of index just get returned as an empty version of the original:

```python
>>> df
   a  b
1  1  4
2  2  5
4  3  6
>>> df[3:2:2].index
Int64Index([], dtype='int64')
```